packages/shell-api/src/helpers.spec.ts

```diff
@@ -181,6 +181,21 @@ describe('getPrintableShardStatus', function () {
     expect(status.databases[0].database._id).to.equal('config');
   });
 
+  describe('hides all internal deprecated fields in shardingVersion', function () {
+    for (const hiddenField of [
+      'minCompatibleVersion',
+      'currentVersion',
+      'excluding',
+      'upgradeId',
+      'upgradeState',
+    ]) {
+      it(`does not show ${hiddenField} in shardingVersion`, async function () {
+        const status = await getPrintableShardStatus(configDatabase, false);
```

Comment:
Very much an optional suggestion, `getPrintableShardStatus()` does quite a lot of work, it might make sense to run it in a `before()` hook inside this `describe()` block (since it returns the same result for all of the tests)?

Reply:
```js
describe.only('hides all internal deprecated fields in shardingVersion', async function () {
    const status = await getPrintableShardStatus(configDatabase, false);

    for (const hiddenField of [
      'minCompatibleVersion',
      'currentVersion',
      'excluding',
      'upgradeId',
      'upgradeState',
    ]) {
      it(`does not show ${hiddenField} in shardingVersion`, async function () {
        expect(status.shardingVersion[hiddenField]).to.equal(undefined);
      });
    }
  });
```

Does not work as describe functions can not be async, so I'll need to keep the getPrintableShardStatus inside the `it`. If you are ok with it, I'll keep it as it is, as it seems to be more complicated than it should be 😞 .

Reply:
Yeah, all good then :+1: